mxcubeweb/core/server/limiter.py

```diff
@@ -0,0 +1,23 @@
+from flask import jsonify
+from flask_limiter import Limiter
+from flask_limiter.util import get_remote_address
+
+
+def rate_limit_error_handler(e):
+    """Handler for rate limit errors"""
+    return jsonify({"error": "Too many requests", "message": str(e.description)}), 429
+
+
+def init_limiter(app):
+    """Initialize the rate limiter"""
+    limiter = Limiter(
+        app=app,
+        key_func=get_remote_address,
+        default_limits=app.config.get(
+            "RATELIMIT_DEFAULT", "150000 per day;6000 per hour"
```

Comment:
Would something like that make sense?

```suggestion
            "RATELIMIT_DEFAULT", configmodels.FlaskConfigModel.RATELIMIT_DEFAULT.split(";")
```

In order to avoid the repetition.

Reply:
The problem is that **RATELIMIT_DEFAULT** and **configmodels.FlaskConfigModel.RATELIMIT_DEFAULT** both come from configmodels.
If there is an issue with configmodels and the value cannot be read, the rate limiter will not initialize because there is no value available. The hardcoded (and repeated) value is just a safety fallback in case this happens.